include/realtime_tools/realtime_box.h

```diff
@@ -26,50 +26,402 @@
+    func(value_);
+  }
+
+  /**
+   * @brief get the content and wait until the mutex could be locked (RealtimeBox behaviour)
+   * @return copy of the value
+   */
+  template <typename U = T>
+  [[nodiscard]] typename std::enable_if_t<!is_ptr_or_smart_ptr<U>, U> get() const
+  {
+    std::lock_guard<mutex_t> guard(lock_);
+    return value_;
+  }
+  /**
+   * @brief get the content and wait until the mutex could be locked
+   * @note same signature as in the existing RealtimeBox<T>
+   */
+  template <typename U = T>
+  typename std::enable_if_t<!is_ptr_or_smart_ptr<U>, void> get(T & in) const
+  {
+    std::lock_guard<mutex_t> guard(lock_);
+    // cppcheck-suppress missingReturn
+    in = value_;
+  }
+  /**
+   * @brief access the content (r) and wait until the mutex could be locked
+   * @note only safe way to access pointer type content (r)
+   * @note same signature as in the existing RealtimeBox<T>
+   */
+  void get(const std::function<void(const T &)> & func)
+  {
+    std::lock_guard<mutex_t> guard(lock_);
+    func(value_);
+  }
+
+  /**
+   * @brief provide a custom assignment operator for easier usage
+   * @note only to be used from non-RT!
+   */
+  template <typename U = T>
+  typename std::enable_if_t<!is_ptr_or_smart_ptr<U>, void> operator=(const T & value)
+  {
+    set(value);
+  }
 
-  Strongly suggested that you use an std::shared_ptr in this box to
-  guarantee realtime safety.
+  /**
+   * @brief provide a custom conversion operator
+   * @note Can only be used from non-RT!
+   */
+  template <typename U = T, typename = typename std::enable_if_t<!is_ptr_or_smart_ptr<U>>>
+  [[nodiscard]] operator T() const
+  {
+    // Only makes sense with the getNonRT method otherwise we would return an std::optional
+    return get();
+  }
+  /**
+   * @brief provide a custom conversion operator
+   * @note Can be used from non-RT and RT contexts
+   */
+  template <typename U = T, typename = typename std::enable_if_t<!is_ptr_or_smart_ptr<U>>>
+  [[nodiscard]] operator std::optional<T>() const
+  {
+    return tryGet();
+  }
+
+  // In case one wants to actually use a pointer
+  // in this implementation we allow accessing the lock directly.
+  // Note: Be careful with lock.unlock().
+  // It may only be called from the thread that locked the mutex!
+  [[nodiscard]] const mutex_t & getMutex() const { return lock_; }
+  [[nodiscard]] mutex_t & getMutex() { return lock_; }
 
- */
-template <class T>
-class RealtimeBox
+private:
+  T value_;
+  mutable mutex_t lock_;
+};
+
+/**
+ * @brief Specialisation for pointer types.
+ * WHY is this specialised. We do not want to break compatibility but show a deprecation note
+ * for get/set etc. methods if used with a pointer type
+ * They are unsafe to use and should therefore be replace with their correspondents that take an std::function for accessing
+ * the value behind the pointer
+*/
+template <class T, typename mutex_type>
+class RealtimeBox<T, mutex_type, true>
```

Comment:
```suggestion
template <class T, typename mutex_type>
[[deprecated("RealtimeBox is not recommended to use with Raw pointers! Use smart pointers instead.")]]
class RealtimeBox<T, mutex_type, true>
```

Reply:
It should actually be: Only use the "get/set(std::function) overloads of the RealtimeBox (which is the reason why we added this template specialisation for pointers).

But that deprecation message is already printed when the default get/set methods are used with pointers!

Reply:
Perhaps I should give a quick recap why it is designed like this given that quite some time has passed now:

1. The existing RealtimeBox implementation is not safe to use with any kind of pointer. Because only any operation on the pointer itself would be locked by the mutex but not the access to the data behind the pointer 

2. The RealtimeBoxBestEffort solves this issue by providing additional overloads to get/set which take a function. These overloads lock the mutex and call the function. Inside this function manipulating the pointer is safe. 
Additionally if the Type is a pointer the normal get/set methods which return a reference to the internal value are disabled via SFINEA (std::enable_if)

3. It makes sense to replace the existing RealtimeBox with the RealtimeBoxBestEffort (This is what this PR aims for). Additionally the requirement was made: We shall not break existing code, but we want to display deprecation notices if the default get/set methods are used with pointers. Therefore I added a template specialisation for pointers removes the std::enable_if for get/set with pointer types and adds deprecation notices for the get/set methods without the std::function overloads. 

This should encourage users to migrate to the get/set(std::function) version for pointer types. 
So in a future release the pointer specialisation of the class can be removed. 

So to answer your question:

Pointer types are supported BUT you need to use the get/set(std::function) methods.

Reply:
@firesurfer I understand your point. What I'm saying is, we don't have to break the Raw pointer in this version, but in the future versions, we can make the users use the smart pointers. So, if we mark the class deprecated with the notice to use the smart pointers in future releases, the users will be able to migrate in the meantime. 

If we mark the methods alone, they still think the Raw pointers will be supported in the future. I'm not sure, if it is interesting to support Raw pointer in the future. If we still want to support, then the way you marked the depreciation is good enough @bmagyar @christophfroehlich any comments on this?